pkg/relayer/proxy/synchronous.go

```diff
@@ -306,10 +307,34 @@ func (sync *synchronousRPCServer) serveHTTP(
 	// request paths and query parameters.
 	// Use the same method, headers, and body as the original request to query the
 	// backend URL.
-	relayHTTPRequest.Host = serviceConfig.BackendUrl.Host
+	requestUrl.Host = serviceConfig.BackendUrl.Host
+
+	// Create the HTTP headers for the request by converting the RelayRequest's
+	// POKTHTTPRequest.Header to an http.Header.
+	headers := http.Header{}
+	for key, headerValues := range poktHTTPRequest.Header {
+		for _, value := range headerValues.Values {
+			headers.Add(key, value)
+		}
+	}
```

Comment:
I've seen this nested loop a couple of times. You have a comment here but not elsewhere.

Do you think there's an opportunity for a helper so it's just one line everywhere?

Reply:
Added a `CopyToHTTPHeader` helper function in the `shannon-sdk` repo.